cmd/scheduler/main.go

```diff
@@ -1,21 +1,31 @@
 package main
 
 import (
+	"fmt"
 	"os"
 
 	apiserver "k8s.io/apiserver/pkg/server"
 	"k8s.io/component-base/logs"
 
+	"github.com/spf13/cobra"
+
 	"github.com/karmada-io/karmada/cmd/scheduler/app"
 )
 
 func main() {
+	stopChan := apiserver.SetupSignalHandler()
+	command := app.NewSchedulerCommand(stopChan)
+	code := run(command)
+	os.Exit(code)
+}
+
+func run(cmd *cobra.Command) int {
```

Comment:
Hi, @kerthcet , do you fix it like this [https://github.com/kubernetes/kubernetes/pull/104503](https://github.com/kubernetes/kubernetes/pull/104503) to replace function `run` with `runSchedulerCmd`, and set `SilenceErrors: true` and `SilenceUsage:  true` to avoid repeatedly print Errors.

Reply:
Are you requesting changes? I leave LGTM to @lonelyCZ 
/approve

Reply:
As you say, we set `SilenceErrors: true` to avoid repeat error printing, but we didn't print errors here. So it's not needed.WDYT?

Reply:
Sorry, I just saw this reply.

I just test it, and it seems to print error twice.
```
[root@master67 karmada]# ./karmada-scheduler
I0124 19:20:16.021823   16059 scheduler.go:64] karmada-scheduler version: version.Info{GitVersion:"v0.7.0-932-g4089758-dirty", GitCommit:"4089758d3029be2b1b813337582714b196868a24", GitTreeState:"dirty", BuildDate:"2022-01-20T20:33:27Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
W0124 19:20:16.022031   16059 client_config.go:615] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
W0124 19:20:16.022052   16059 client_config.go:620] error creating inClusterConfig, falling back to default config: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined
Error: error building kubeconfig: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
Usage:
  karmada-scheduler [flags]
  ......
Use "karmada-scheduler [command] --help" for more information about a command.

command failed: error building kubeconfig: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
```

Reply:
I suggest to change it to

```
func main() {
	if err := runSchedulerCmd(); err != nil {
		// fmt.Fprintf(os.Stderr, "%v\n", err)
		os.Exit(1)
	}
}

func runSchedulerCmd() error {
	logs.InitLogs()
	defer logs.FlushLogs()

	stopChan := apiserver.SetupSignalHandler()

	command := app.NewSchedulerCommand(stopChan)

	if err := command.Execute(); err != nil {
		return err
	}

	return nil
}
```

This looks a little clearer. How do you think?

Reply:
The duplicate error is printed by `fmt.Println` here, removed, thanks for caching.
I have no strong preference of the code implement, updated already. Thanks.